jujugui/static/gui/src/app/app.js

```diff
@@ -872,6 +868,11 @@ YUI.add('juju-gui', function(Y) {
       credentials, useMacaroons, apis=[this.controllerAPI, this.env]) {
       if (useMacaroons) {
         apis.forEach(api => {
+          if (credentials && api.name === 'model-api') {
```

Comment:
This is done twice in this method can you extract it out into a function? It can even be local to this function.

Why is this done twice now instead of only done where it was before? Can you add a comment as to why this is necessary? If I recall this was done so that the u/p entered in the `Login` component would get added to the user class so this block of it would never evaluate true?